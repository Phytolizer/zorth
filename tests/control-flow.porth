include "io.porth"

// if/else
34 35 + 70 = if 69 print else 420 print end
34 35 + 69 = if 69 print else 420 print end

"------------------------------\n" write

// while
10 while dup 0 > do
  dup print
  1 -
end
