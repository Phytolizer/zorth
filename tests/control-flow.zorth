// Prepare the separation line
0 while dup 30 < do
  dup mem + 45 .
  1 +
end
mem + 10 .

// if/else
34 35 + 70 = if 69 print else 420 print end
34 35 + 69 = if 69 print else 420 print end

31 mem 1 1 syscall3 drop

// while
10 while dup 0 > do
  dup print
  1 -
end
