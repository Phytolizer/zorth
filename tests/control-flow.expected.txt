: returncode 0
: stdout 59
420
69
------------------------------
10
9
8
7
6
5
4
3
2
1

: stderr 0

